airflow-core/src/airflow/api_fastapi/common/parameters.py

```
@@ -185,6 +185,53 @@ def depends(cls, *args: Any, **kwargs: Any) -> Self:
         raise NotImplementedError("Use search_param_factory instead , depends is not implemented.")
 
 
+class QueryTaskInstanceTaskGroupFilter(BaseParam[str]):
+    """Task group filter - returns all tasks in the specified group."""
+
+    def __init__(self, dag=None, skip_none: bool = True):
+        super().__init__(skip_none=skip_none)
+        self._dag: None | Any = dag
+
+    @property
+    def dag(self) -> None | Any:
+        return self._dag
+
+    @dag.setter
+    def dag(self, value: None | Any) -> None:
+        self._dag = value
+
+    def to_orm(self, select: Select) -> Select:
+        if self.value is None and self.skip_none:
+            return select
+
+        if not self.dag:
+            raise ValueError("DAG must be set before calling to_orm")
+
+        if not hasattr(self.dag, "task_group"):
+            return select
+
+        matching_task_ids = []
+
+        # Exact matching on group_id
```

Not sure should we raise Not Found error if the task_group not found like:

https://github.com/apache/airflow/blob/7c6fcecc535de447482b18c85fb98235655e745d/airflow-core/src/airflow/api_fastapi/execution_api/routes/task_instances.py#L950-L976